src/openapi-generator.ts

```diff
@@ -853,7 +862,9 @@ export class OpenAPIGenerator {
     return omitBy(metadata, isNil);
   }
 
-  private getMetadata(zodSchema: ZodSchema<any>) {
+  private getMetadata<T extends any>(
```

Comment:
Does inferring the type help with anything here? I am okay with this change, just asking if it is cosmetic or if it helped with a TS usecase.

Reply:
I added it to have proper types here, otherwise the example would be `any` type.
![image](https://user-images.githubusercontent.com/12901858/196109144-124acba3-6b70-4a1f-a04e-2c0d55549bc5.png)

I will probably not need it once I updated the PR but I think it's a good addition nevertheless. But I can also revert it if you want to.